src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/Grammar/GrammarGenerator.cs

```diff
@@ -74,17 +74,23 @@ public static string Run(List<TreeType> types)
                 }
             }
 
+            // Add some rules not present in Syntax.xml.
+            rules.Add("Utf8StringLiteralToken", [Join(" ", [RuleReference("StringLiteralToken"), RuleReference("Utf8Suffix")])]);
+            rules.Add("Utf8MultiLineRawStringLiteralToken", [Join(" ", [RuleReference("MultiLineRawStringLiteralToken"), RuleReference("Utf8Suffix")])]);
+            rules.Add("Utf8SingleLineRawStringLiteralToken", [Join(" ", [RuleReference("SingleLineRawStringLiteralToken"), RuleReference("Utf8Suffix")])]);
+            rules.Add("Utf8Suffix", [new("'u8'"), new("'U8'")]);
+
             // The grammar will bottom out with certain lexical productions. Create rules for these.
             var lexicalRules = rules.Values.SelectMany(ps => ps).SelectMany(p => p.ReferencedRules)
                 .Where(r => !rules.TryGetValue(r, out var productions) || productions.Count == 0).ToArray();
             foreach (var name in lexicalRules)
-                rules[name] = new List<Production> { new Production("/* see lexical specification */") };
+                rules[name] = [new("/* see lexical specification */")];
 
             var seen = new HashSet<string>();
 
             // Define a few major sections to help keep the grammar file naturally grouped.
             var majorRules = ImmutableArray.Create(
-                "CompilationUnitSyntax", "MemberDeclarationSyntax", "TypeSyntax", "StatementSyntax", "ExpressionSyntax", "XmlNodeSyntax", "StructuredTriviaSyntax");
+                "CompilationUnitSyntax", "MemberDeclarationSyntax", "TypeSyntax", "StatementSyntax", "ExpressionSyntax", "XmlNodeSyntax", "StructuredTriviaSyntax", "Utf8Suffix");
```

Comment:
ensures we write out all the utf8 rules first before printing out the utf8 suffix.